### PR TITLE
NCD - Creatinine, Pregnancy and Liver Function labs

### PR DESCRIPTION
**Pull request by anvmn**
#418
#420
#421

### CODE REVIEW

**Comment by anvmn** [comment]
Peek 2022-09-06 16-27

**Comment by anvmn** [comment]
Peek 2022-09-06 21-35

**Comment by anvmn** [comment]
@AronNovak please review

File: client/src/elm/Measurement/Utils.elm [hunk 1]
```diff
@@ -2818,3 +2961,856 @@ laboratoryTaskIconClass task =
...
+
+
+resultFormHeaderSection : Language -> NominalDate -> Maybe NominalDate -> LaboratoryTask -> List (Html msg)
+resultFormHeaderSection language currentDate executionDate task =
+    let
+        executionDateSection =
+            Maybe.map
+                (\date ->
+                    [ viewLabel language <| Translate.LaboratoryTaskDate task
+                    , p [ class "test-date" ] [ text <| formatDDMMYYYY date ]
```
Comment (AronNovak) [inline client/src/elm/Measurement/Utils.elm]:
```suggestion
                    , p [ class "execution-date" ] [ text <| formatDDMMYYYY date ]
```
Perhaps match the naming with other parts

File: client/src/elm/Pages/GlobalCaseManagement/View.elm [hunk 2]
```diff
@@ -939,24 +934,127 @@ viewPrenatalLabsEntry :
...
+            generateNCDLabsEntries language currentDate filteredItemsDict db
+
+        content =
+            if List.isEmpty entries then
+                [ translateText language Translate.NoMatchesFound ]
+
+            else
+                List.map (viewNCDLabsEntry language) entries
+    in
+    div [ class "pane" ]
```
Comment (AronNovak) [inline client/src/elm/Pages/GlobalCaseManagement/View.elm:1003]:
```suggestion
    div [ class "pane ncd-labs" ]
```

File: client/src/elm/Pages/NCD/Activity/Utils.elm [hunk 3]
```diff
@@ -57,10 +57,13 @@ expectActivity currentDate assembled db activity =
         MedicalHistory ->
             True
 
-        -- @todo
-        _ ->
+        Laboratory ->
             True
 
+        NextSteps ->
+            -- @todo
```
Comment (AronNovak) [inline client/src/elm/Pages/NCD/Activity/Utils.elm:64]: Do we have the ability to sort it out now?

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      -- | SetActiveNextStepsTask NextStepsTask
+      -- | SetReferralBoolInput (Bool -> ReferralForm -> ReferralForm) Bool
+      -- | SetFacilityNonReferralReason (Maybe ReasonForNonReferral) ReferralFacility ReasonForNonReferral
+      -- | SaveSendToHC PersonId (Maybe ( NCDSendToHCId, NCDSendToHC )) (Maybe NextStepsTask)
+      -- | SetMedicationDistributionBoolInput (Bool -> MedicationDistributionForm -> MedicationDistributionForm) Bool
+      -- | SetMedicationDistributionAdministrationNote (Maybe AdministrationNote) MedicationDistributionSign AdministrationNote
+      -- | SetRecommendedTreatmentSign (List RecommendedTreatmentSign) RecommendedTreatmentSign
+      -- | SaveMedicationDistribution PersonId (Maybe ( NCDMedicationDistributionId, NCDMedicationDistribution )) (Maybe NextStepsTask)
+      -- | SetHealthEducationBoolInput (Bool -> HealthEducationForm -> HealthEducationForm) Bool
+      -- | SaveHealthEducation PersonId (Maybe ( NCDHealthEducationId, NCDHealthEducation )) (Maybe NextStepsTask)
```
Comment (AronNovak) [inline client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:58]:
```suggestion
```

Shall we get rid of the commented out code or is it kind of a todo?

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    | CloseLabsResultsEntry PersonId NCDLabsResultsId LabsResultsValue
+
+
+type alias Model =
+    { labResultsData : LabResultsData
+
+    -- , nextStepsData : NextStepsData
```
Comment (AronNovak) [inline client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:65]: same here

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+--     }
+--
+--
+-- emptyNextStepsData : NextStepsData
+-- emptyNextStepsData =
+--     { referralForm = emptyReferralForm
+--     , medicationDistributionForm = emptyMedicationDistributionForm
+--     , healthEducationForm = emptyHealthEducationForm
+--     , activeTask = Nothing
+--     }
```
Comment (AronNovak) [inline client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:111]: same here

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+    -- , nextStepsData : NextStepsData
+    }
+
+
+emptyModel : Model
+emptyModel =
+    { labResultsData = emptyLabResultsData
+
+    -- , nextStepsData = emptyNextStepsData
```
Comment (AronNovak) [inline client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:73]: and here

File: client/src/elm/Pages/NCD/RecurrentActivity/Update.elm [hunk 8]
```diff
@@ -0,0 +1,705 @@
...
+update language currentDate id db msg model =
+    let
+        noChange =
+            ( model, Cmd.none, [] )
+
+        generateLabResultsMsgs nextTask =
+            Maybe.map (\task -> [ SetActiveLabResultsTask task ]) nextTask
+                |> Maybe.withDefault [ SetActivePage <| UserPage <| NCDRecurrentEncounterPage id ]
+
+        -- generateNextStepsMsgs personId nextTask =
```
Comment (AronNovak) [inline client/src/elm/Pages/NCD/RecurrentActivity/Update.elm:43]: Likely todo or eliminate

File: client/src/elm/Pages/NCD/RecurrentActivity/Update.elm [hunk 9]
```diff
@@ -0,0 +1,705 @@
...
+                            )
+                        |> Maybe.withDefault []
+            in
+            ( model
+            , Cmd.none
+            , appMsgs
+            )
+                |> sequenceExtra (update language currentDate id db) extraMsgs
+
+        -- SetActiveNextStepsTask task ->
```
Comment (AronNovak) [inline client/src/elm/Pages/NCD/RecurrentActivity/Update.elm:465]: todo or eliminate ?

File: client/src/elm/Pages/GlobalCaseManagement/View.elm (hunk 2 above)
Comment (anvmn) [reply client/src/elm/Pages/GlobalCaseManagement/View.elm:1003]: Not required for styling, and other types ofpanes do add extra class, so I'd rather leave it as is.

File: client/src/elm/Pages/NCD/Activity/Utils.elm (hunk 3 above)
Comment (anvmn) [reply client/src/elm/Pages/NCD/Activity/Utils.elm:64]: @AronNovak NCD Next steps related development is done at #456 and #459, All commented out code is handled there.

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm (hunk 4 above)
Comment (anvmn) [reply client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:58]: Next steps related development is done at https://github.com/TIP-Global-Health/eheza-app/pull/456 and https://github.com/TIP-Global-Health/eheza-app/pull/459, All commented out code is handled there.

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm (hunk 5 above)
Comment (anvmn) [reply client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:65]: Next steps related development is done at https://github.com/TIP-Global-Health/eheza-app/pull/456 and https://github.com/TIP-Global-Health/eheza-app/pull/459, All commented out code is handled there.

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm (hunk 6 above)
Comment (anvmn) [reply client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:111]: Next steps related development is done at https://github.com/TIP-Global-Health/eheza-app/pull/456 and https://github.com/TIP-Global-Health/eheza-app/pull/459, All commented out code is handled there.

File: client/src/elm/Pages/NCD/RecurrentActivity/Model.elm (hunk 7 above)
Comment (anvmn) [reply client/src/elm/Pages/NCD/RecurrentActivity/Model.elm:73]: Next steps related development is done at https://github.com/TIP-Global-Health/eheza-app/pull/456 and https://github.com/TIP-Global-Health/eheza-app/pull/459, All commented out code is handled there.

File: client/src/elm/Pages/NCD/RecurrentActivity/Update.elm (hunk 8 above)
Comment (anvmn) [reply client/src/elm/Pages/NCD/RecurrentActivity/Update.elm:43]: Next steps related development is done at https://github.com/TIP-Global-Health/eheza-app/pull/456 and https://github.com/TIP-Global-Health/eheza-app/pull/459, All commented out code is handled there.

File: client/src/elm/Pages/NCD/RecurrentActivity/Update.elm (hunk 9 above)
Comment (anvmn) [reply client/src/elm/Pages/NCD/RecurrentActivity/Update.elm:465]: Next steps related development is done at https://github.com/TIP-Global-Health/eheza-app/pull/456 and https://github.com/TIP-Global-Health/eheza-app/pull/459, All commented out code is handled there.

**Pull request merged**